src/runtime/meter-toast-api-test.js

```diff
@@ -341,6 +342,7 @@ describes.realWin('MeterToastApi', (env) => {
     activitiesMock.expects('openIframe').resolves(port);
     const messageStub = sandbox.stub(port, 'execute');
     await meterToastApi.start();
+    await tick(1);
```

Comment:
I'm curious, why did the test need to be updated?

Reply:
I had updated the `ActivityIframeView`'s `whenComplete` method to be `async` and return `Promise<void>`, since nothing was using its result. By making that function `async` though, it meant a rejection was happening an event loop later. I modified the tests so they would one more event loop for the error behavior to happen.

I just updated the code to remove the `async` modifier, which means the tests don't need to be updated anymore. It does mean the `whenComplete` method returns something that the app code isn't using though. I'm cool with either approach. What do you think?

Reply:
Ah, I see, this is because of the type of `whenComplete` in `View`. I don't think the abstract method should be declared to return `ActivityResult`, `View` is meant to be general and should not be specific to activity results. Can we make ActivityIframeView return Promise<void> to match the interface, but not change it to async to avoid changing any behavior?

Reply:
In this case, I think using `async` would improve the code. If we're just `await`ing the result without returning, then async/await would express that pretty clearly. End users wouldn't notice a difference. The tests would just need those `tick()`s

Reply:
I reverted the code so that `whenComplete` has an `async` modifier and return `Promise<void>`. This is the best approach I can think of currently, and it doesn't change the behavior for end users. If you have a specific suggestion that would make the code more readable, I'm definitely interested in hearing it.

Reply:
OK, SG, thanks.